src/Simplex/Messaging/Agent/Store/SQLite.hs

```diff
@@ -29,109 +29,66 @@ import UnliftIO.STM
---       sndQ <- getSndQueue st sndQId
---       return $ SomeConn SCDuplex (DuplexConnection connAlias rcvQ sndQ)
---     (Just rcvQId, _) -> do
---       rcvQ <- getRcvQueue st rcvQId
---       return $ SomeConn SCReceive (ReceiveConnection connAlias rcvQ)
---     (_, Just sndQId) -> do
---       sndQ <- getSndQueue st sndQId
---       return $ SomeConn SCSend (SendConnection connAlias sndQ)
---     _ -> throwError SEBadConn
-
--- getReceiveQueue :: SQLiteStore -> SMPServer -> SMP.RecipientId -> m (ConnAlias, ReceiveQueue)
--- getReceiveQueue st SMPServer {host, port} recipientId = do
---   rcvQueue <- getRcvQueueByRecipientId st recipientId host port
---   connAlias <- getConnAliasByRcvQueue st recipientId
---   return (connAlias, rcvQueue)
-
--- -- TODO make transactional
--- addSndQueue :: SQLiteStore -> ConnAlias -> SendQueue -> m ()
--- addSndQueue st connAlias sndQueue =
---   getConn st connAlias
---     >>= \case
---       SomeConn SCDuplex _ -> throwError (SEBadConnType CDuplex)
---       SomeConn SCSend _ -> throwError (SEBadConnType CSend)
---       SomeConn SCReceive _ -> do
---         srvId <- upsertServer st (server (sndQueue :: SendQueue))
---         sndQ <- insertSndQueue st srvId sndQueue
---         updateRcvConnectionWithSndQueue st connAlias sndQ
-
--- -- TODO make transactional
--- addRcvQueue :: SQLiteStore -> ConnAlias -> ReceiveQueue -> m ()
--- addRcvQueue st connAlias rcvQueue =
---   getConn st connAlias
---     >>= \case
---       SomeConn SCDuplex _ -> throwError (SEBadConnType CDuplex)
---       SomeConn SCReceive _ -> throwError (SEBadConnType CReceive)
---       SomeConn SCSend _ -> do
---         srvId <- upsertServer st (server (rcvQueue :: ReceiveQueue))
---         rcvQ <- insertRcvQueue st srvId rcvQueue
---         updateSndConnectionWithRcvQueue st connAlias rcvQ
-
--- -- TODO think about design of one-to-one relationships between connections ans send/receive queues
--- -- - Make wide `connections` table? -> Leads to inability to constrain queue fields on SQL level
--- -- - Make bi-directional foreign keys deferred on queue side?
--- --   * Involves populating foreign keys on queues' tables and reworking store
--- --   * Enables cascade deletes
--- --   ? See https://sqlite.org/foreignkeys.html#fk_deferred
--- -- - Keep as is and just wrap in transaction?
--- deleteConn :: SQLiteStore -> ConnAlias -> m ()
--- deleteConn st connAlias = do
---   (rcvQId, sndQId) <- getConnection st connAlias
---   forM_ rcvQId $ deleteRcvQueue st
---   forM_ sndQId $ deleteSndQueue st
---   deleteConnection st connAlias
---   when (isNothing rcvQId && isNothing sndQId) $ throwError SEBadConn
+      deleteConnCascade dbConn connAlias
+
+  upgradeConnWithSndQueue :: SQLiteStore -> ConnAlias -> SendQueue -> m ()
+  upgradeConnWithSndQueue SQLiteStore {dbConn} connAlias sndQueue =
+    liftIO (updateRcvConnWithSndQueue dbConn connAlias sndQueue) >>= \case
+      Right () -> return ()
+      Left e -> throwError e
```

Comment:
I think it's liftEither

Reply:
Changed and pushed it, but after looking at it again wasn't happy about it. It's just immediately clear what the function returns w/t having to look at its type signature when using case - decided to revert the change